### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by hyork03** [cagnulein/qdomyos-zwift#1376]
**Describe the bug**
Peloton app sees QZ but does not connect

**To Reproduce**
1. Turn on bike and HR Monitor
2. Turn on iPad and open peloton app 
3. Open QZ on iPhone
4. Try to connect to QZ
5. See error message “unable to connect to QZ”

**Expected behavior**
it usually connects right away 

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone6] iPhone SE
 - OS: [e.g. iOS8.1] iOS 15.6.1

**Append a debug log**
[debug-Fri_Mar_24_09_13_04_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/11062738/debug-Fri_Mar_24_09_13_04_2023.log)

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
 
**Additional context**

**Comment by cagnulein** [comment]
hi @hyork03 the main issue about the log is that you have the "Cycling Cadence Sensor (Peloton compatibility)" setting disabled in the peloton settings in QZ

Let me know!

**Comment by cagnulein** [comment]
hi yes that setting must be on for the peloton compatibility, if it was off (as before) you need to redo a network reset on the ipad and a reboot (since the cache is now different from the actual qz behaviour).
Also I don't see the log because you need to upload it from github directly
Thanks!

**Comment by hyork03** [comment]
Ok I put that setting on in the QZ app in my phone. Then quit the app, network reset my iPad, and tried to connect. It still recognizes QZ but won’t connect. Do you want me to upload a bug log?

**Comment by cagnulein** [comment]
yes please, upload on github directly. It must be something else

**Comment by hyork03** [comment]
[debug-Fri_Mar_24_11_26_01_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/11063852/debug-Fri_Mar_24_11_26_01_2023.log)

**Comment by cagnulein** [comment]
@hyork03 everything is ok in the log. you can try several things:
1) try to install nrfconnect on the ipad, and with qz open, check how qz is showned on the nrfconnect app (send me a screenshot here). You should be able to connect to it
2) you can try to swap the devices (qz on ipad and peloton on iphone) just to test if there is still something wrong on the ipad.

Let me know

**Comment by hyork03** [comment]
Hi there, after more trial and error I was finally able to fix this issue – there was something wrong with the Bluetooth connection between my iPad and QZ - It wasit was acknowledging the device but not connecting. I deleted all of the Bluetooth connections on my iPad and then opened the Bluetooth settings on iPad and then opened the QZ app - the iPad recognized the new connection and requested to connect and now it is working again. Thanks for your efforts to try to solve it!

**Comment by cagnulein** [comment]
Glad to hear this!
